ddtrace/tracer/civisibility_payload.go

```diff
@@ -0,0 +1,117 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024 Datadog, Inc.
+
+package tracer
+
+import (
+	"bytes"
+	"sync/atomic"
+
+	"github.com/tinylib/msgp/msgp"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/globalconfig"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/version"
+)
+
+// civisibilitypayload represents a payload specifically designed for CI Visibility events.
+// It embeds the generic payload structure and adds methods to handle CI Visibility specific data.
+type civisibilitypayload struct {
+	*payload
+}
+
+// push adds a new CI Visibility event to the payload buffer.
+// It grows the buffer to accommodate the new event, encodes the event in MessagePack format, and updates the event count.
+//
+// Parameters:
+//
+//	event - The CI Visibility event to be added to the payload.
+//
+// Returns:
+//
+//	An error if encoding the event fails.
+func (p *civisibilitypayload) push(event *ciVisibilityEvent) error {
+	p.buf.Grow(event.Msgsize())
+	if err := msgp.Encode(&p.buf, event); err != nil {
+		return err
+	}
+	atomic.AddUint32(&p.count, 1)
+	p.updateHeader()
+	return nil
+}
+
+// newCiVisibilityPayload creates a new instance of civisibilitypayload.
+//
+// Returns:
+//
+//	A pointer to a newly initialized civisibilitypayload instance.
+func newCiVisibilityPayload() *civisibilitypayload {
+	return &civisibilitypayload{newPayload()}
+}
+
+// GetBuffer retrieves the complete body of the CI Visibility payload, including metadata.
+// It reads the current payload buffer, adds metadata, and encodes the entire payload in MessagePack format.
+//
+// Parameters:
+//
+//	config - A pointer to the config structure containing environment settings.
+//
+// Returns:
+//
+//	A pointer to a bytes.Buffer containing the encoded CI Visibility payload.
+//	An error if reading from the buffer or encoding the payload fails.
+func (p *civisibilitypayload) GetBuffer(config *config) (*bytes.Buffer, error) {
```

Comment:
`civisibilitypayload` isn't exported. Do you need `GetBuffer` to be exported?

Reply:
you're right, fixed in [465456a](https://github.com/DataDog/dd-trace-go/pull/2739/commits/465456aeed42aa1b9a6b4a96c0dad5a402e09308)